### PR TITLE
Refactor table URL filtering for testability

### PR DESCRIPTION
**Pull request by jpellizzari**
Re-work the DataTable URL logic to make it easily testable.

This also sneaks in a change to add the source to the ReconciliationGraph as I was pairing with @joshri  and I forgot to switch branches. It is in it's own commit, so keeping it in the PR.

### CODE REVIEW

File: ui/components/AutomationsTable.tsx [hunk 1]
```diff
@@ -122,18 +119,17 @@ function AutomationsTable({ className, automations, hideSource }: Props) {
...
-      ),
+      value: () => <Timestamp time={""} />,
     },
   ];
 
   if (hideSource) fields = _.filter(fields, (f) => f.label !== "Source");
 
   return (
     <FilterableTable
+      initialSelections={{ "type:Kustomization": true }}
```
Comment (joshri) [inline ui/components/AutomationsTable.tsx:130]: These prop changes need to happen on every FilterableTable - aka sources and reconciled objects - right now those tables don't crash but also don't change the url

**Comment by jpellizzari** [comment]
Originally outlined in #2054

File: ui/components/FilterableTable.tsx [hunk 2]
```diff
@@ -192,11 +265,13 @@ function FilterableTable({
...
     });
-    setUrl(resetFormState);
+    doChange(resetFormState);
+    // setUrl(resetFormState);
   };
 
   const handleFilterSelect = (filters, formState) => {
-    setUrl(formState);
+    doChange(formState);
+    // setUrl(formState);
```
[the remark this answers is not in the document]
Comment (jpellizzari) [reply ui/components/FilterableTable.tsx]: Removed

File: ui/components/__tests__/FilterableTable.test.tsx [hunk 3]
```diff
@@ -543,4 +546,92 @@ describe("FilterableTable", () => {
...
+    expect(tableRows[0].innerHTML).toContain("foo");
+  });
+  it("returns a query string on filter change", () => {
+    const initialFilterConfig = {
+      ...filterConfigForString(rows, "type"),
+    };
+
+    const recorder = jest.fn();
+    const handler = (sel: FilterSelections) => {
+      const queryString = filterSelectionsToQueryString(sel);
```
[the remark this answers is not in the document]
Comment (jpellizzari) [reply ui/components/__tests__/FilterableTable.test.tsx]: Removed

File: ui/components/FilterDialog.tsx [hunk 4]
```diff
@@ -107,14 +91,14 @@ const FilterSection = ({ header, options, formState, onSectionSelect }) => {
...
+  onFilterSelect: (val: FilterConfig, state: FilterSelections) => void;
   /** Object containing column headers + corresponding filter options */
   filterList: FilterConfig;
-  formState: DialogFormState;
+  formState: FilterSelections;
   open?: boolean;
 }
 
-export function formStateToFilters(values: DialogFormState): FilterConfig {
+export function selectionsToFilters(values: FilterSelections): FilterConfig {
```
[the remark this answers is not in the document]
Comment (jpellizzari) [reply ui/components/FilterDialog.tsx:101]: @joshri cannot reproduce. Can you double check you have a clean version of this branch?

File: ui/components/AutomationsTable.tsx (hunk 1 above)
Comment (jpellizzari) [reply ui/components/AutomationsTable.tsx]: Yeah this was just a debug thing. Nice catch. Removed.

File: ui/components/FilterDialog.tsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (joshri) [reply ui/components/FilterDialog.tsx:101]: Rebooted everything and it's lookin good - was probs happening on my end

**Review by joshri — APPROVED** [review]
ReconciledObjectsTable uses a FilterableTable which should be changed to URLAdressableTable 🌞  🌞

**Comment by jpellizzari** [comment]
> ReconciledObjectsTable uses a FilterableTable which should be changed to URLAdressableTable sun_with_face sun_with_face

Should it? I dunno if EVERY table is URL addressable...In any case we can decide if we want to add it

**Pull request merged**